File: 02/code/objc/Lection02/Lection02/HelloViewController.m
//
//  HelloViewController.m
//  Lection02
//
//  Created by d.taraev on 16.03.17.
//  Copyright © 2017 mail.ru. All rights reserved.
//

#import "HelloViewController.h"

@interface HelloViewController ()
@property (weak, nonatomic) IBOutlet UILabel *label;
@end

@implementation HelloViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
    
    self.label.text = self.name ? [NSString stringWithFormat:@"Скоро концерт, %@, пойдем", self.name] : nil;
}

@end

File: 02/code/swift/Lection02/HelloViewController.swift
//
//  HelloViewController.swift
//  Lection02
//
//  Created by Gennady Evstratov on 11/03/2017.
//  Copyright © 2017 Mail. All rights reserved.
//

import UIKit

class HelloViewController: UIViewController {

    var name: String?
    
    @IBOutlet weak var label: UILabel?
    
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        
        guard let name = name else {
            label?.text = nil
            return
        }
        
        label?.text = "Скоро концерт, \(name), пойдем"
    }
}
